addons/project/tests/test_task_dependencies.py

```diff
@@ -103,32 +103,128 @@ def test_tracking_dependencies(self):
         self.flush_tracking()
+        self.assertEqual(len(self.task_1.message_ids), 1,
+            'Changing the deadline on task 1 should have logged a message in task 1.')
         self.assertEqual(len(self.task_2.message_ids), 1,
             'Changing the deadline on task 1 should not have logged a message in task 2.')
 
         # Check that changing a field that is not tracked at all on task 2 does not impact task 1.
         self.task_2.color = 100 # no new message
         self.flush_tracking()
-        self.assertEqual(len(self.task_1.message_ids), 2,
+        self.assertEqual(len(self.task_1.message_ids), 1,
             'Changing the color on task 2 should not have logged a message in task 1 since it is not tracked.')
 
-        # Check that changing multiple fields does not log more than one message.
-        self.task_2.write({
-            'date_deadline': date(2020, 1, 1),
-            'kanban_state': 'blocked',
-        }) # + 1 message in task_1 and task_2
+    def test_tracking_stage_dependencies(self):
+        """
+            this test ensure that the stage modification from blocking tasks are correctly sending notifications to their parent when necessary.
+            When all blocking task of a task are in a folded stage, log a notification.
+            When a blocking tasking is reopened, and all the other blocking tasks of the parents are closed, log a notification.
+        """
+        # Enable the company setting
+        self.env['res.config.settings'].create({
+            'group_project_task_dependencies': True
+        }).execute()
+        # adds stages to the project
+        stages = dev, validation, done, cancel = self.env['project.task.type'].create([{
+            'sequence': 1,
+            'name': 'Dev',
+            'fold': False,
+        }, {
+            'sequence': 2,
+            'name': 'Validation',
+            'fold': False,
+        }, {
+            'sequence': 29,
+            'name': 'Done',
+            'fold': True,
+        }, {
+            'sequence': 30,
+            'name': 'Cancel',
+            'fold': True,
+        }])
+        self.project_pigs.write(
+            {'type_ids': [Command.link(stage_id) for stage_id in stages.ids]})
+        child_tasks = child_task_1, child_task_2, child_task_3 = self.env['project.task'].create([{
+            'name': 'child_task_%s' % i,
+            'user_ids': self.user_projectuser,
+            'project_id': self.project_pigs.id,
+            'stage_id': dev.id,
+        } for i in range(1, 4)])
+        parent_task_1, parent_task_2, parent_task_3 = self.env['project.task'].create([{
+            'name': 'parent_task_%s' % value[0],
+            'user_ids': self.user_projectuser,
+            'project_id': self.project_pigs.id,
+            'stage_id': dev.id,
+            'depend_on_ids': value[1],
+        } for value in [[1, child_task_1], [2, child_tasks], [3, child_tasks]]])
```

Comment:
`for i, task in enumerate(child_tasks)` will do the trick. If 'name' really has to start from 1, just `'parent_task_%s' % (i + 1)`

Reply:
yes, i'd like to keep the 1 to 3, so i dont have to check again all the error log message of the asserts. :D